pkg/common/resource.go

```diff
@@ -99,12 +103,38 @@ func getResource(resourceList v1.ResourceList) *si.Resource {
 	return resources.Build()
 }
 
-func CreateUpdateRequestForTask(appID, taskID string, resource *si.Resource) si.UpdateRequest {
+func createTagsForTask(pod *v1.Pod, taskID string) map[string]string {
+	tags := map[string]string{
+		"namespace": pod.Namespace,
+		"podName": pod.Name,
+	}
+
+	// add Pod labels to Task tags
+	labels, err := json.Marshal(pod.Labels)
+	if err != nil {
+		log.Logger.Debug("Could not marshal pod labels", zap.String("taskID", taskID))
+	} else {
+		tags["labels"] = string(labels)
+	}
+
+	// add Pod annotations to Task tags
+	annotations, err := json.Marshal(pod.Annotations)
```

Comment:
This is dangerous: annotations have no limits on what they can contain as a value. They can contain large blobs of data. I am not sure that we just want to pull that in without some kind of vetting. 

I also doubt that we can do much with it when we get it into the scheduler.

Reply:
Removed.